src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +327,201 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Checks if summary tag present.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean containsSummaryTag(DetailNode javadoc) {
+        final DetailNode node = getFirstInlineTag(javadoc);
+        return node != null && isSummaryTag(node);
+    }
+
+    /**
+     * Finds and return first inline tag node from a javadoc root node.
+     *
+     * @param javadoc Javadoc root node.
+     * @return First inline tag node or null if no node is found.
+     */
+    private static DetailNode getFirstInlineTag(DetailNode javadoc) {
+        DetailNode node = null;
+        final DetailNode[] children = javadoc.getChildren();
+        for (DetailNode child: children) {
+            // If present as a children of javadoc
+            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                node = child;
+            }
+            // If nested inside html tag
+            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                node = getInlineTagNodeWithinHtmlElement(child);
+            }
+
+            if (node != null) {
+                break;
+            }
+        }
+        return node;
+    }
+
+    /**
+     * Returns inline javadoc tag node inside html tags from a HTML tag.
```

Comment:
This is a bit confusing; would `Returns an inline javadoc tag node that is within a HTML tag.` be more clear?

Reply:
Changed to your suggestion. Done

Reply:
Resolved.